atomicfu/src/nativeInterop/cinterop/posixparking.def

```diff
@@ -0,0 +1,40 @@
+headers = stdlib.h stdint.h pthread.h
+---
+typedef struct {
+    pthread_mutex_t mutex;
+    pthread_cond_t cond;
+    uint64_t wake;
+} posix_combo_t;
+
+posix_combo_t* posixParkInit() {
+    posix_combo_t* pc = malloc(sizeof(posix_combo_t));
+    pthread_mutex_init(&pc->mutex, NULL);
+    pthread_cond_init(&pc->cond, NULL);
+    pc->wake = 0;
+    return pc;
+}
+
+void posixWait(posix_combo_t* pc) {
+    pthread_mutex_lock(&pc->mutex);
+    while(!pc->wake) {
+        pthread_cond_wait(&pc->cond, &pc->mutex);
+    }
+    pthread_mutex_unlock(&pc->mutex);
+    pthread_mutex_destroy(&pc->mutex);
+    pthread_cond_destroy(&pc->cond);
+    free(pc);
+}
+
+void posixWake(posix_combo_t* pc) {
+    pthread_mutex_lock(&pc->mutex);
+    pc->wake = 1;
+    pthread_cond_signal(&pc->cond);
+    pthread_mutex_unlock(&pc->mutex);
+}
+
+void posixDestroy(posix_combo_t* pc) {
+    pthread_mutex_destroy(&pc->mutex);
+    pthread_cond_destroy(&pc->cond);
+    free(pc);
+}
```

Comment:
Alright that is good to know!

The reason why I used cinterop is because the provided posix library in kotlin/native is slightly different over different platforms. 
For instance, on windows you use `pthread_cond_tVar` and on linux `pthread_cond_t`. Also the `pthread_cond_init `function is not available in at the nativeMain sourceset. 
So instead of having to duplicate this over the platforms, I thought this would be the cleanest solution.

Reply:
Yep, the function API differences are why https://github.com/Kotlin/kotlinx-atomicfu/pull/440/files had to duplicate code and couldn't write it in nativeMain. That's a pity, but what can you do.